.github/ISSUE_TEMPLATE/release_template.md

```diff
@@ -0,0 +1,65 @@
+---
+name: Release Checklist
+about: Use this template for upcoming releases.
+title: "[Release Number] Release Checklist"
+---
+
+Target date:
+
+Release Candidate by YYYY-MM-DD
+
+Release by YYYY-MM-DD
+
+Release candidate
+-------------------
+- [ ] Verify that all relevant PRs have been merged to master.
+- [ ] Create a PR against master to bump version number, merge that PR
+- [ ] From the commit just before bumping the version, create a new branch `maint/<release version number>`
+- [ ] Update changelog and open PR targeting a new `maint/<release version number>` branch
```

Comment:
This is where we might want to mention the `etstool.py changelog build` command? There is a mention in https://github.com/enthought/apptools/blob/4f4178c86ca0393555715245b6ac5047b1434cad/etstool.py#L73-L80 but it is a bit hidden. The release checklist might be a better place for at least a mention. We probably want to make sure the used news fragment are deleted in this step? If so, let's mention it too.

(For later, we may revisit if we want to update the changelog build command such that the clean action is done by default rather than optionally by prompt.)